api/v1/_submissions/PKPBackendSubmissionsHandler.inc.php

```diff
@@ -67,109 +76,129 @@ public function authorize($request, &$args, $roleAssignments)
     }
 
     /**
-     * Get a list of submissions according to passed query parameters
+     * Get a collection of submissions
      *
      * @param $slimRequest Request Slim request object
      * @param $response Response object
+     * @param array $args arguments
      *
      * @return Response
      */
     public function getMany($slimRequest, $response, $args)
     {
-        $request = $this->getRequest();
+        $request = Application::get()->getRequest();
         $currentUser = $request->getUser();
         $context = $request->getContext();
 
-        // Merge query params over default params
-        $defaultParams = [
-            'count' => 20,
-            'offset' => 0,
-        ];
+        if (!$context) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        $collector = $this->getSubmissionCollector($slimRequest->getQueryParams());
+
+        HookRegistry::call('API::submissions::params', [$collector, $slimRequest]);
 
-        // Anyone not a manager or site admin can only access their assigned
-        // submissions
-        $userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+        // Prevent users from viewing submissions they're not assigned to,
+        // except for journal managers and admins.
+        $userRoles = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_USER_ROLES);
         $canAccessUnassignedSubmission = !empty(array_intersect([ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER], $userRoles));
-        if (!$canAccessUnassignedSubmission) {
-            $defaultParams['assignedTo'] = [$currentUser->getId()];
+        if (!$canAccessUnassignedSubmission && !in_array($currentUser->getId(), $collector->assignedTo)) {
```

Comment:
We could change it to verify if `getSubmissionCollector` is empty? Because `$canAccessUnassignedSubmission` it's being implemented inside the `getSubmissionCollector` too.

Reply:
You're right! This is duplicated unnecessarily. Good catch! We probably need to keep the one here, because it is fired after the hook. This prevents a custom plugin from accidentally permitting access to unassigned submissions.